benchmarking-cli/src/main.rs

```diff
@@ -87,46 +86,52 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
     // let's read the program
     let program_string = fs::read_to_string(format!("../examples/{}.masm", &args.example))?;
 
-    // let's read the input fils
+    // let's read the input files
     let input_string = fs::read_to_string(format!("../examples/{}.inputs", &args.example))
         .map_err(|err| format!("Failed to open input file `{}` - {}", &args.example, err))?;
     let inputs_des: InputFile = serde_json::from_str(&input_string)
         .map_err(|err| format!("Failed to deserialize input data - {}", err))
         .unwrap();
 
-    let stack_init = get_stack_init(&inputs_des);
-    let advice_tape = get_advice_tape(&inputs_des);
-
-    let inputs = get_program_inputs(&stack_init, &advice_tape);
+    let advice_provider = parse_advice_provider(&inputs_des).unwrap();
+    let stack_input = parse_stack_inputs(&inputs_des).unwrap();
 
     // Compilation time
     let now = Instant::now();
-    let assembler = Assembler::new();
+    let assembler = Assembler::default();
     let program = assembler
         .compile(&program_string)
         .expect("Could not compile source");
 
     println! {"Compilation Time (cold): {} ms", now.elapsed().as_millis()}
 
+    let program_hash = program.hash();
+    let kernel = Kernel::default();
+    let program_info = ProgramInfo::new(program_hash, kernel);
+
     let now = Instant::now();
     let _program2 = assembler
         .compile(&program_string)
         .expect("Could not compile source");
 
     println! {"Compilation Time (hot): {} ms", now.elapsed().as_millis()}
 
+
+    let stack_input_cloned = stack_input.clone();
```

Comment:
I am unsure if I need to `clone()` those. In Miden v0.3 I was able to borrow but that seems not possible anymore.

Reply:
It seems the API changed and now a clone is required.